### PR TITLE
Move internal/x/reflect to internal/reflect

### PR DESCRIPTION
**Pull request by peter-edge**

### CODE REVIEW

File: internal/reflect/handler.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+func newHandler(options ...HandlerOption) *handler {
+	handler := &handler{
+		logger: zap.NewNop(),
+	}
+	for _, option := range options {
+		option(handler)
+	}
+	// TODO(pedge): composition
```
Comment (AlexAPB) [inline internal/reflect/handler.go]: can we drop this todo?

Comment (peter-edge) [reply internal/reflect/handler.go]: We can, but I still want to do it...so... :-)

Comment (AlexAPB) [reply internal/reflect/handler.go]: haha, then let's do it?!

Comment (peter-edge) [reply internal/reflect/handler.go]: I need to do this in like three places...perhaps this should be `// TODO: composition?` like as in a question lol

Comment (AlexAPB) [reply internal/reflect/handler.go]: lol sounds good, let's merge it

**Pull request merged**